arches/app/models/querysets/graph.py

```diff
@@ -9,7 +12,20 @@ def create(self, *args, **kwargs):
             "Use create_graph() to create new Graph instances with proper business logic."
         )
 
-    def create_graph(self, name="", user=None, is_resource=False):
+    def get_slug(self, name, is_resource):
+        if name:
+            slug = slugify(name, separator="_")
+        else:
+            if is_resource:
+                slug = "new_resource_model"
+            else:
+                slug = "new_branch"
+        existing_slugs = GraphModel.objects.all().values_list("slug", flat=True)
```

Comment:
self is already a queryset, which may have already been filtered. not likely that it's been filtered here, but still good to do IMO in case people copy this pattern for other methods.
```suggestion
        existing_slugs = self.values_list("slug", flat=True)
```